app/src/main/java/com/rentminder/service/HouseholdService.kt

```diff
@@ -9,11 +9,18 @@ import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
 class HouseholdService {
-    suspend fun fetchHouseholdsList() : List<Household>?{
+    private val service: IHouseholdDAO by lazy {
+        RetrofitClientInstance.retrofitInstance?.create(IHouseholdDAO::class.java)
+            ?: throw IllegalStateException("RetrofitClientInstance is null")
+    }
```

Comment:
Implemented Lazy here to initialize the RetrofitClientInstance. This is so its only initialized one time, instead of every time the 'fetchHouseholdList()' is called.